lib/pulfalight/traject/ead2_config.rb

```diff
@@ -252,6 +253,218 @@
 
 to_field "descrules_ssm", extract_xpath("/ead/eadheader/profiledesc/descrules")
 
+compose "physical_holdings", lambda { |record, accumulator, _context|
+  elements = record.xpath("//dsc[@type='othertype']/*[is_component(.)]", NokogiriXpathExtensions.new)
+  accumulator.concat(elements)
+} do
+
+  to_field "box_number_ssi" do |record, accumulator|
+    unitid_element = record.at_xpath("./did/container[@type='box']")
+    accumulator << unitid_element&.text&.to_i
+  end
+  to_field "box_number_ssm" do |_record, accumulator, context|
+    box_numbers = context.output_hash.fetch("box_number_ssi", [])
+
+    accumulator.concat box_numbers.map(&:to_s)
+  end
+
+  to_field "barcode_ssi" do |record, accumulator|
+    unitid_element = record.at_xpath("./did/unitid[@type='barcode']")
+    accumulator << unitid_element&.text&.to_i
+  end
+  to_field "barcode_ssm" do |_record, accumulator, context|
+    barcodes = context.output_hash.fetch("barcode_ssi", [])
+
+    accumulator.concat barcodes.map(&:to_s)
+  end
+
+  to_field "physical_location_code_ssm" do |record, accumulator|
+    unitid_element = record.at_xpath("./did/physloc[@type='code']")
+    accumulator << unitid_element&.text
+  end
+
+  to_field "physical_location_ssm" do |_record, accumulator, context|
+    location_codes = context.output_hash.fetch("physical_location_code_ssm", [])
+    location_code = location_codes.first
+    next unless location_code
+
+    location = Pulfalight::PhysicalLocationResolver.resolve(location_code)
+    accumulator << location
+  end
+
+  # This is for indexing components
```

Comment:
I am still struggling with recursion here within the scope of Traject configuration. I have attempted one approach in which the I invoke the following:

```ruby
to_field "components" do |record, accumulator, context|
    child_components = record.xpath("./*[is_component(.)]", NokogiriXpathExtensions.new)
    child_components.each do |child_component|
      previous_parent = context.clipboard[:parent]
      context.clipboard[:parent] = record
      output = map_record(child_component)
      accumulator << output
      context.clipboard[:parent] = previous_parent
    end
  end
```

...as this approach was proposed in #218. However, I find that, when `map_record` in invoked within the scope of a `compose` block, the indexing steps *outside* of this block can no longer be run.

Perhaps through metaprogramming I can dynamically define indexing steps for fields attempt this approach for recursion (I avoided attempting this given the time consumed by difficulties in recursion I already experienced).